tests/virtual_machines_test.go

```diff
@@ -100,6 +100,30 @@ var _ = Describe("[rfe_id:3503][crit:medium][vendor:cnv-qe@redhat.com][level:com
 				}
 			})
 
+			Context("and the client tries to assign the same MAC address for two different vm, one created with dryRun=All mode", func() {
+				It("should success creating a vm with the same mac allocated to a dry run", func() {
+					By("Creating a vm with dry run mode")
+					vm := CreateVmObject(TestNamespace, false, []kubevirtv1.Interface{newInterface("br", "")}, []kubevirtv1.Network{newNetwork("br")})
+					createOptions := &client.CreateOptions{}
+					client.DryRunAll.ApplyToCreate(createOptions)
+
+					err := testClient.VirtClient.Create(context.TODO(), vm, createOptions)
+					Expect(err).ToNot(HaveOccurred())
+					Expect(net.ParseMAC(vm.Spec.Template.Spec.Domain.Devices.Interfaces[0].MacAddress)).ToNot(BeEmpty(), "Should successfully parse mac address")
+
+					vmDryRunOverlap := CreateVmObject(TestNamespace, false, []kubevirtv1.Interface{newInterface("brDryRunOverlap", "")}, []kubevirtv1.Network{newNetwork("brDryRunOverlap")})
```

Comment:
This should be the actual test, and we should add a `By("creating a vm with the same mac to make sure the dryRun assigned mac is not occupied on the macpool")`